src/ICSharpCode.SharpZipLib/Core/FileSystemScanner.cs

```diff
@@ -496,31 +389,29 @@ private void ScanDir(string directory, bool recurse)
 				}
 			}
 
-			if (alive_ && recurse)
+			if (!alive_ || !recurse) return;
+			
+			try
 			{
-				try
+				var subDirectories = GetDirectories(directory)
+					.Where(d => d != null && directoryFilter_.IsMatch(d));
```

Comment:
Check `direcotryFilter_` for null. Perhaps by
```suggestion
				Func<bool, string> filter = directoryFilter_ == null 
					? (d) => true 
					: (d) =>  directoryFilter_.IsMatch(d);
				var subDirectories = GetDirectories(directory)
					.Where(d => d != null && filter(d));
```